include/openPMD/Series.hpp

```diff
@@ -53,34 +53,80 @@ namespace openPMD
 {
 class ReadIterations;
 class WriteIterations;
+class Series;
+class SeriesImpl;
 
-/** @brief  Root level of the openPMD hierarchy.
+namespace internal
+{
+/**
+ * @brief Data members for Series. Pinned at one memory location.
+ *
+ * (Not movable or copyable)
+ *
+ */
+class SeriesData : public AttributableData
+{
+    friend class openPMD::SeriesImpl;
+    friend class openPMD::Iteration;
+    friend class openPMD::Series;
+
+public:
+    explicit SeriesData() = default;
+
+    SeriesData( SeriesData const & ) = delete;
+    SeriesData( SeriesData && ) = delete;
+
+    SeriesData & operator=( SeriesData const & ) = delete;
+    SeriesData & operator=( SeriesData && ) = delete;
+
+    virtual ~SeriesData() = default;
+
+    Container< Iteration, uint64_t > iterations{};
+
+OPENPMD_private :
+    /**
+     *  Whether a step is currently active for this iteration.
+     * Used for group-based iteration layout, see SeriesData.hpp for
+     * iteration-based layout.
+     * Access via stepStatus() method to automatically select the correct
+     * one among both flags.
+     */
+    std::shared_ptr< StepStatus >
```

Comment:
Memo to ourselves: as mentioned in the PR description, those `shared_ptr` members will become regular members